src/main/java/com/mongodb/hibernate/jdbc/MongoConnectionProvider.java

```diff
@@ -82,7 +82,7 @@ public final class MongoConnectionProvider implements ConnectionProvider, Config
     @Serial
     private static final long serialVersionUID = 1L;
 
-    private @Nullable MongoClient mongoClient;
+    private transient @Nullable MongoClient mongoClient;
```

Comment:
Making `mongoClient` transient means that `MongoConnectionProvider` serializes and deserializes deceivingly successfully, but the deserialized object is invalid, which is why I don't think it's the right approach, and we should remove `transient`.

To solve this, `MongoConnectionProvider` could

1. store the configuration supplied via the `configure` method in an instance field;
2. serialize it instead of `mongoClient`, assuming that all the property values can be serialized;
3. when being deserialized, re-initialize `mongoClient` based on the deserialized configuration.

However, I failed to find any explanation of why `Service` is `Serializable`. If we ever discover why that is, and realize that `MongoConnectionProvider` should actually be serializable, we may try the approach described above. But for now, I think we should not do it.

We are left with two options:
a) Do nothing, which results in the following exception when serializing `MongoConnectionProvider`: `java.io.NotSerializableException: com.mongodb.client.internal.MongoClientImpl`.
b) Fail explicitly, with a clarifying message:

```java
@Serial
private void writeObject(ObjectOutputStream out) throws IOException {
    throw new NotSerializableException(
            "This class is not designed to be serialized despite it having to implement `Serializable`");
}
```
Adding this method to `MongoConnectionProvider` results in the following exception when serializing `MongoConnectionProvider`: `java.io.NotSerializableException: This class is not designed to be serialized despite it having to implement Serializable`.

My proposal is to use the approach b).

Reply:
I did ask Hibernate code author but currently he is busy without having time to respond.
I adopted the proposed option.